app/javascript/lib/scp-api.js

```diff
@@ -0,0 +1,138 @@
+ * 
+ * @param {Boolean} mock Whether to use mock data.  Helps development, tests.
+ * @returns {Promise} Promise object described in "Example return" above
+ */
+export async function fetchAuthCode(mock=false) {
+  return await scpApi('/search/auth_code', mock);
+}
+
+/**
+ * Returns a list of all available search facets, including default filter values
+ * 
+ * Docs: https:///singlecell.broadinstitute.org/single_cell/api/swagger_docs/v1#!/Search/search_facets_path
+ * 
+ * @param {Boolean} mock Whether to use mock data.  Helps development, tests.
+ * @returns {Promise} Promise object containing camel-cased data from API
+ */
+export async function fetchFacets(mock=false) {
+  return await scpApi('/search/facets', mock);
+}
+
+// If true, returns mock data for all API responses.  Only for dev.
+let globalMock = false;
+
+/**
+ * Sets flag on whether to use mock data for all API responses.
+ *
+ * This method is useful for tests and certain development scenarios,
+ * e.g. when evolving a new API or to work around occasional API blockers.
+ *
+ * @param {Boolean} flag Whether to use mock data for all API responses
+ */
+export function setGlobalMockFlag(flag) {
+  globalMock = flag;
+}
+
+// Modifiable in setMockOrigin, used in unit tests
+let mockOrigin = '';
+
+/**
+ * Sets origin (e.g. http://localhost:3000) for mocked SCP API URLs
+ *
+ * This enables mock data to be used from Jest tests
+ *
+ * @param {Boolean} origin Origin (e.g. http://localhost:3000) for mocked SCP API URLs
+ */
+export function setMockOrigin(origin) {
+  mockOrigin = origin;
+}
+
+/**
+ * Returns a list of matching filters for a given facet
+ * 
+ * Docs: https:///singlecell.broadinstitute.org/single_cell/api/swagger_docs/v1#!/Search/search_facet_filters_path
+ * 
+ * @example
+ * // returns Promise for mock JSON in /mock_data/facets_filters_disease_tuberculosis.json
+ * fetchFacetsFilters('disease', 'tuberculosis', true);
+ * // returns Promise for live JSON as shown example from "Docs" link above (but camel-cased)
+ * fetchFacetsFilters('disease', 'tuberculosis');
+ * @param {String} facet Identifier of facet
+ * @param {String} query User-supplied query string
+ * @param {Boolean} mock Whether to use mock data.  Helps development, tests.
+ * @returns {Promise} Promise object containing camel-cased data from API
+ */
+export async function fetchFacetsFilters(facet, query, mock=false) {
+  
+  const queryString = (!(mock || globalMock)) ? `?facet=${facet}&query=${query}` : `_${facet}_${query}`;
+
+  return await scpApi(`/search/facets_filters${queryString}`, mock);
+}
+
+/**
+ * Client for SCP REST API.  Less fetch boilerplate, easier mocks.
+ * 
+ * @param {String} path | Relative path for API endpoint, e.g. /search/auth_code
+ * @param {Boolean} mock | Whether to use mock data.  Helps development, tests.
+ */
+export default async function scpApi(path, mock=false) {
```

Comment:
this function will eventually need to accept an optional POST body, right? so you might as well put that argument in now to avoid rewrites later.  Also, now that I think more about it, I'm not sure this will be able to handle GETs and POSTs to the same URL, so you might just want to add a 'method' argument too.

Reply:
Addressed in 99e334a.

The new general `init` object parameter, described in more detail in my other reply, provides a way to handle `method` and all other fetch settings.